Service cache docs (#2347)
esl/MongooseIM: doc/advanced-configuration/Services.md

    * `stanza`: Adds `send_message_chat`, `send_message_headline`, `send_stanza_c2s`
    * `stats`: Adds `stats`, `stats_host`
    * `vcard`: Adds `get_vcard`, `get_vcard2`, `get_vcard2_multi`, `set_vcard`, `set_vcard2`, `set_vcard2_multi`
    * `gdpr`: Adds `retrieve_personal_data`

### Example configuration
` {service_admin_extra, [{submods, [node, accounts, sessions]}]} `

> Module `service_admin_extra` depends on `service_cache`(see below) as the cache speeds up GDPR data retrieval and removal.

Reviewer: ```suggestion
The `service_admin_extra` module depends on the`service_cache`(see below) as the cache speeds up GDPR data retrieval and removal.
```

> Module `service_admin_extra` depens on `service_cache`(see below) as cache is efficiently used for `gdpr` submodule and GDPR compilant account removal.

Reviewer 2: depens -> depends

Reviewer 2: `as the cache speeds up GDPR data retrieval and removal.`?

### service_cache

> The implementation of cache storage.

Reviewer: ```suggestion
Implementation of the cache storage.
```
not sure which one sounds better

> Implementation of cache storage.

Reviewer 3: The implementation

> The Implementation of cache storage.

Reviewer 2: Implementation -> implementation

> Module exposes one function `lookup/2` with parameters `Key` and `Fun`.

Reviewer: ```suggestion
The module exposes a function `lookup/2` with parameters `Key` and `Fun`.
```

> Module expose one function `lookup/2` with parameters `Key` and `Fun`.

Reviewer 3: module exposes one

> When called and the key `Key` exists in cache, the related value is returned.

Reviewer 3: when the function is called
